### ISSUE TITLE
Reduce image size

### ISSUE THREAD

**Issue by orzechow** [watzon/docker-nerdfonts#1]
Thanks @watzon for this repo, I like the idea :+1: 

I tried pulling the docker image, but it's still very large though (>4GB) – despite using `alpine:latest` and removing unnecessary packages.

That's because every `RUN` command creates a new layer in the Docker image. 
Even though you remove packages like `curl`, etc, the layers where they were installed still exist in the image history, contributing to its size.

To reduce the size, you can use a [multi-stage build](https://docs.docker.com/build/building/multi-stage/) approach. The idea is to download and extract the fonts in one stage and copy only the necessary files to a minimal `scratch` (empty base image) stage:

```dockerfile
FROM alpine:latest

# Install required packages for downloading and extracting
RUN apk add --no-cache bash curl jq tar xz

# Copy the download script
COPY scripts/download-fonts.sh /usr/local/bin/
RUN chmod +x /usr/local/bin/download-fonts.sh

# Download and extract fonts
RUN /usr/local/bin/download-fonts.sh


# Next Stage
FROM scratch

# Copy fonts from the builder stage
COPY --from=builder /usr/share/fonts/nerd-fonts /usr/share/fonts/nerd-fonts

# Set working directory to fonts location
WORKDIR /usr/share/fonts/nerd-fonts

# This container is meant to be used as a source for fonts
CMD ["ls", "-la"]
```

As a result, the last stage will only contain the fonts folder.
Your GitHub Action will only push the last stage to ghcr.io (if I'm not mistaken :wink:  )

**Comment by watzon** [comment]
Love it. Would you like to make a PR?

**Comment by orzechow** [comment]
Nice that you like it 😁

> Would you like to make a PR?

I can do so, but not today anymore.

**Comment by orzechow** [comment]
Okay, I had a second look.
Turns out, the dominating factor are not these few apk packages (I already wondered).
Instead, the fonts themself are huge and you provide them all.
- 70 fonts
- multiple `.ttf` files (font styles) per font
- some of these are >5MB each (IosevkaNerdFont even 13MB per `.ttf`)

I guess you should rethink serving all fonts in all variants in one image.
Maybe one image variant per font? Like `docker-nerdfonts:RobotoMono` etc.

**Comment by watzon** [comment]
That's definitely an idea. It would just take exponentially more time to build all of them. Tbh I mostly made this for my [Inkify](https://github.com/watzon/inkify) project which needs all of the fonts.

**Comment by orzechow** [comment]
Yeah, probably not worth the effort. Thanks anyways, I found another solution for my project.
